### PR TITLE
Added SetWithTemplate, FormatTemplate and ConcatFormat

### PR DESCRIPTION
**Pull request by spearmin10**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://github.com/demisto/content/pull/21420

## Description
Build a text from a template text.

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by ShacharKidor** [comment]
Hi @spearmin10,
Thank you for your contribution!
Your PR review will be slightly delayed as we are at the start of the holiday season -
I apologize in advance.

In the meantime, please fill in the registration form:
> Thank you for your contribution. Your generosity and caring are unrivaled! Make sure to register your contribution by filling the [Contribution Registration](https://forms.gle/XDfxU4E61ZwEESSMA) form, so our content wizard @ShacharKidor will know he can start review the proposed changes.


Thanks again :)

**Comment by spearmin10** [comment]
@ShacharKidor This is a new PR which fixed some issues and changed some design to https://github.com/demisto/content/pull/21420. Please close the old PR.

**Comment by ShacharKidor** [comment]
> @ShacharKidor This is a new PR which fixed some issues and changed some design to #21420. Please close the old PR.

@spearmin10 - Sure. Thank you :)

**Comment by spearmin10** [comment]
@moishce Could you please review the PR?

**Comment by moishce** [comment]
@spearmin10 Sure, I'll do that in the next few days.
I apologize for the delay.
the holiday season slowed things down

**Comment by bziser** [comment]
Hi @spearmin10 
Thank you for your contribution! Good work :)

I'm reviewing your PR, could you please add a short recorded demo for the formatters?

Please feel free to reach out to me with any questions - I'm available here or on slack :)
Thanks again

File: Packs/CommonScripts/Scripts/SetWithTemplate/SetWithTemplate.py [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+
+class ContextData:
+    def __init__(self, context: Optional[Dict[str, Any]] = None):
+        self.__context = context or demisto.context()
+
+    def __get_inputs_value(self, key: str) -> Any:
+        inputs = demisto.callingContext.get('context.PlaybookInputs') or {}
```
Comment (bziser) [inline Packs/CommonScripts/Scripts/SetWithTemplate/SetWithTemplate.py:12]:
```suggestion
        inputs = demisto.callingContext.get('context.PlaybookInputs', {})
```

File: Packs/CommonScripts/Scripts/SetWithTemplate/SetWithTemplate.py [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+        return demisto.dt(inputs, key)
+
+    def __get_incident_value(self, key: str) -> Any:
+        incident = demisto.incident()
+        val = demisto.dt(incident, key)
+        if val is None:
+            val = demisto.dt(incident.get('CustomFields') or {}, key)
```
Comment (bziser) [inline Packs/CommonScripts/Scripts/SetWithTemplate/SetWithTemplate.py:19]:
```suggestion
            val = demisto.dt(incident.get('CustomFields', {}), key)
```

**Review by bziser — COMMENTED** [review]
These are my initial comments.
Please change in all your files the use of the `get` function. 
You can pass to `get` function the default you want it to use if it didn't find value.
As you can see [here](https://www.w3schools.com/python/ref_dictionary_get.asp)

Great work 👍

**Comment by spearmin10** [comment]
@bziser Thank you for revewing and your comments.
It doesn't pass test suites when passing default value for `get`. Because, for example,
When args is `{'stringify': None}`, `args.get('stringify', 'noop')` returns None, not `noop`.

**Comment by bziser** [comment]
@spearmin10 
Hi, you are right. But if some of the `args` are not passed, we are not passing them with None value, we don't pass them at all. 
Please also merge from master and resolve the conflicts

**Comment by idovandijk** [comment]
@spearmin10 how is this different than using the `replace` transformer or using actual DT?

**Comment by efelmandar** [comment]
@bziser finished reviewing the PR, everything's OK from our side

**Comment by bziser** [comment]
@spearmin10  Hi, please merge from master and resolve the conflicts and I'll merge it to internal PR :)

**Comment by spearmin10** [comment]
@bziser Sure, I've updated the conflicts.

**Review by bziser — APPROVED** [review]

**Pull request merged**